Stable merge window for week 52 of 2020

raisjn opened the pull request:
Updated Packages:

* chessmarkable -> 0.5.1-1
* oxide -> 2.0.1~beta-2
* plato -> 0.9.1-11
* retris -> 0.6.1-1
* rmkit
  * harmony -> 0.1.0-3
  * mines -> 0.1.0-2
  * nao -> 0.1.0-2
  * remux -> 0.1.3-1
  * simple -> 0.1.0-2

New Packages:

* rm2fb -> 1.0.0

Commit:
Stable merge window for week 52 of 2020

Updated Packages:

* chessmarkable -> 0.5.1-1
* oxide -> 2.0.1~beta-2
* plato -> 0.9.1-11
* retris -> 0.6.1-1
* rmkit
  * harmony -> 0.1.0-3
  * mines -> 0.1.0-2
  * nao -> 0.1.0-2
  * remux -> 0.1.3-1
  * simple -> 0.1.0-2

New Packages:

* rm2fb -> 1.0.0

raisjn commented:
the major focus of this PR is rM2 compatibility. rm2fb is a new package and all the other updated packages have had changes related to rM2 compatibility.

Eeems commented on package/rm2fb/package:
```diff
@@ -0,0 +1,56 @@
+configure() {
+    systemctl daemon-reload
+    systemctl enable rm2fb --now
+    systemctl restart xochitl
```
hmm, I missed this when rm2fb was initially started. We should only restart it if it's running. Since configure runs on upgrade, we shouldn't be restarting the service if they've disabled it for something else.

raisjn replied:
good point. should the fix be put into stable? (i don't think its high pri, since no one will be upgrading the package until we push an update)

Eeems replied:
I'd prefer to have it in place before going to stable. While it's not likely going to cause issues yet, it does have various unintended side effects that someone might encounter if they are re-enabling entware or reinstalling the package.

Eeems commented on package/rm2fb/rm2fb.service:
```diff
@@ -0,0 +1,20 @@
+
+[Unit]
+Description=Remarkable2 Framebuffer Server
+Before=xochitl.service
```
This doesn't seem like it'll okay well with launchers that intend for tarnish to stay disabled

raisjn replied:
i don't think `Before=` actually causes xochitl to start, it is just a hint how to enable both services if they are both started at same time. (i think)

`Requires=` will cause xochitl (or the listed service) to start (if i understand systemd properly), but it will start them both at same time (hence the Before/After keywords)

Eeems replied:
If it's a hint, likely it should be marked as before the launchers as well?

LinusCDE committed:
Update plato to 0.9.1-12 (rM2 bugfix) (#166)

Eeems committed:
Update rm2fb (#168)

Only restart xochitl if it's running.
Reset crashnum before restarting xochtil.
Modify preremove message based on xochitl's state.

raisjn commented:
tested on rM2:

* rm2fb
  * reinstalled
* chessmarkable
* plato
  * disabling inputs works properly
* oxide
  * can launch apps, correctly redraws screen on app exit
* retris
* harmony
* mines
* nao
* remux

all worked as expected with touch and stylus inputs

Commit:
[rm2fb] fix typo in preremove

Commit:
[rmkit] add missing icons back in
